### ISSUE TITLE
Neither snapshot or livestream work- help!

### ISSUE THREAD

**Issue by rwswan84** [Sunoo/homebridge-camera-ffmpeg#1238]
I've been racking my brain trying to get these streams to work. 

I'm running Homebridge on my Synology nas as a package install. On the same nas I'm running surveillance station with five Swann cameras (which are actually re-branded Hiks) and one Amcrest. They all work perfectly, and can get their RTSP streams in VLC just fine. The RTSP streams are also working in Home Assistant. 

I have multiple RTSP urls to choose from, but I can't get any to work. I even tried the "share stream path" url directly from SS with the same results. 
Examples: 
rtsp://user:pass@192.168.1.191/Streaming/Channels/1
rtsp://user:pass@192.168.1.191/Streaming/Channels/2
rtsp://user:pass@192.168.1.191:554/Streaming/Channels/1
rtsp://user:pass@192.168.1.191:554/Streaming/Channels/2
rtsp://user:pass@192.168.1.191/live/h264
rtsp://user:pass@192.168.1.191:554/live/h264
rtsp://user:pass@192.168.1.191:554/mpeg4
rtsp://user:pass@192.168.1.191/ISAPI/Streaming/channels/101/HighResolutionVideo
rtsp://syno:122cb0231037f700b056b4d29dc8117b@192.168.1.51:554/Sms=6.unicast

All URLS above work in VLC

The snapshot urls also work fine in VLC or in a browser:
http://user:pass@192.168.1.191/cgi-bin/snapshot.cgi
http://user:pass@192.168.1.191/Streaming/Channels/1/picture
rtsp://user:pass@192.168.1.191:554/Sms=6.unicast
http://user:pass@192.168.1.191/Streaming/channels/1/picture?snapShotImageType=JPEG

Below is my current code

        {
            "name": "Camera FFmpeg",
            "cameras": [
                {
                    "name": "Synology - front door",
                    "videoConfig": {
                        "source": "rtsp://user:pass@192.168.1.191/Streaming/Channels/1",
                        "stillImageSource": "http://192.168.1.191/Streaming/Channels/1/picture",
                        "maxStreams": 1,
                        "debug": true
                    }
                }
            ],
            "platform": "Camera-ffmpeg"
        }

I've also tried it with all combination of ffmpeg arguments like -rtsp_transport, -re, -i, and all combinations as such. 

Below is the copy of error log

n(1280,iw)':'min(720,ih)':force_original_aspect_ratio=decrease,scale=trunc(iw/2)*2:trunc(ih/2)*2 -b:v 299k -payload_type 99 -ssrc 3512731 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params bUw+HXWpBQTnL7RkmMbtuiZ+ioqN3XhtsuZyDMnK srtp://192.168.1.114:61590?rtcpport=61590&pkt_size=1316 -loglevel level+verbose -progress pipe:1
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info] ffmpeg version 4.1.6 Copyright (c) 2000-2020 the FFmpeg developers
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   built with gcc 7.5.0 (GCC)
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   configuration: --prefix=/usr --incdir='${prefix}/include/ffmpeg' --arch=i686 --target-os=linux --cross-prefix=/usr/local/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu- --enable-cross-compile --enable-optimizations --enable-pic --enable-gpl --enable-shared --disable-static --disable-stripping --enable-version3 --enable-encoders --enable-pthreads --disable-protocols --disable-protocol=rtp --enable-protocol=file --enable-protocol=pipe --disable-muxer=image2 --disable-muxer=image2pipe --disable-swscale-alpha --disable-ffplay --disable-ffprobe --disable-doc --disable-devices --disable-bzlib --disable-altivec --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libmp3lame --disable-vaapi --disable-cuvid --disable-nvenc --disable-decoder=amrnb --disable-decoder=ac3 --disable-decoder=ac3_fixed --disable-encoder=zmbv --disable-encoder=dca --disable-decoder=dca --disable-encoder=ac3 --disable-encoder=ac3_fixed --disable-encoder=eac3 --disable-decoder=eac3 --disable-encoder=truehd --disable-decoder=truehd --disable-encoder=hevc_vaapi --disable-decoder=hevc --disable-muxer=hevc --disable-demuxer=hevc --disable-parser=hevc --disable-bsf=hevc_mp4toannexb --x86asmexe=yasm --cc=/usr/local/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu-wrap-gcc --enable-yasm --enable-libx264 --enable-encoder=libx264
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libavutil      56. 22.100 / 56. 22.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libavcodec     58. 35.100 / 58. 35.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libavformat    58. 20.100 / 58. 20.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libavdevice    58.  5.100 / 58.  5.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libavfilter     7. 40.101 /  7. 40.101
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libswscale      5.  3.100 /  5.  3.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libswresample   3.  3.100 /  3.  3.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [info]   libpostproc    55.  3.100 / 55.  3.100
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [error] Unrecognized option 'srtp_out_suite'.
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [fatal] Error splitting the argument list: Option not found
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] FFmpeg exited with code: 1 and signal: null (Error)
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] Stopped video stream.

**Comment by Sunoo** [comment]
I don’t know much about Synology, but it looks like your system has a weird (possibly very old) version of FFmpeg that it is trying to use. I’m not familiar enough with that system to tell you where or how to get a more current version though, unfortunately.

**Comment by rwswan84** [comment]
Thanks for getting back to me! 
I believe I got whatever version comes with the current install of homebridge-camera-ffmpeg plugin. 

Here is my ffmpeg details from the terminal
ffmpeg version 4.1.6 Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 7.5.0 (GCC)
  configuration: --prefix=/usr --incdir='${prefix}/include/ffmpeg' --arch=i686 --target-os=linux --cross-prefix=/usr/local/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu- --enable-cross-compile --enable-optimizations --enable-pic --enable-gpl --enable-shared --disable-static --disable-stripping --enable-version3 --enable-encoders --enable-pthreads --disable-protocols --disable-protocol=rtp --enable-protocol=file --enable-protocol=pipe --disable-muxer=image2 --disable-muxer=image2pipe --disable-swscale-alpha --disable-ffplay --disable-ffprobe --disable-doc --disable-devices --disable-bzlib --disable-altivec --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libmp3lame --disable-vaapi --disable-cuvid --disable-nvenc --disable-decoder=amrnb --disable-decoder=ac3 --disable-decoder=ac3_fixed --disable-encoder=zmbv --disable-encoder=dca --disable-decoder=dca --disable-encoder=ac3 --disable-encoder=ac3_fixed --disable-encoder=eac3 --disable-decoder=eac3 --disable-encoder=truehd --disable-decoder=truehd --disable-encoder=hevc_vaapi --disable-decoder=hevc --disable-muxer=hevc --disable-demuxer=hevc --disable-parser=hevc --disable-bsf=hevc_mp4toannexb --x86asmexe=yasm --cc=/usr/local/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu-wrap-gcc --enable-yasm --enable-libx264 --enable-encoder=libx264
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

Can you make any sense of these errors?
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [error] Unrecognized option 'srtp_out_suite'.
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] [fatal] Error splitting the argument list: Option not found
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] FFmpeg exited with code: 1 and signal: null (Error)
[12/20/2021, 12:23:07 AM] [Camera FFmpeg] [Synology - front door] Stopped video stream.

The srtp ip it's trying to connect to is my iphone where I'm trying to load the video stream thru HomeKit/Homebridge.

**Comment by Movieaholic** [comment]
I'm having a similar if not the same problem if i remove and re-add the cameras they work as normal until tonight when i did it now just the snapshots refresh as the used to.  last time I did it they went back to fully functional for less that 12 hours then reverted to always saying now (for the refresh time) and being totally black or having a old snapshot photo but no video stream if i click on them its the same for Unifi Protect and Ring so im not sure exactly whats going on.

**Comment by Sunoo** [comment]
@Movieaholic There is a known issue on Apple’s side right now with snapshots not working properly. This happens even with real HomeKit cameras.

**Comment by Snoomy** [comment]
> I believe I got whatever version comes with the current install of homebridge-camera-ffmpeg plugin.
> Here is my ffmpeg details from the terminal ffmpeg version 4.1.6 Copyright (c) 2000-2020 the FFmpeg developers built with gcc 7.5.0 (GCC)

@rwswan84, the problem is homebridge-camera-ffmpeg plugin uses ffmpeg, that's been preinstalled on your Synology NAS. I've just faced the same problem like you after updating to DSM 7 and switching from homebridge that had been running inside Docker container under DSM 6 to a native DSM 7 homebridge app.

So the solution you need is to manually update your outdated FFmpeg 4.1.6 to the latest 4.4.1

Here you can get a static FFmpeg build: https://www.johnvansickle.com/ffmpeg/
I have a Synology DS918+, which is Intel based, so I had to use amd64 version. Choose the one appropriate to your Synology NAS device. And here is the FAQ in case you are not much familiar with shell: https://www.johnvansickle.com/ffmpeg/faq/

Long story short - you need to turn on ssh in you Synology device's settings, connect with admin credentials and run the following commands (relevant for Intel-based devices on 6th of January 2021):
$ wget https://johnvansickle.com/ffmpeg/builds/ffmpeg-4.4.1-amd64-static.tar.xz
$ tar xvf ffmpeg-release-amd64-static.tar.xz
$ sudo mv /usr/bin/ffmpeg ./
$ sudo cp ./ffmpeg-4.4.1-amd64-static/ffmpeg /usr/bin/
$ chmod 755 /usr/bin/ffmpeg

**Comment by rwswan84** [comment]
> > I believe I got whatever version comes with the current install of homebridge-camera-ffmpeg plugin.
> > Here is my ffmpeg details from the terminal ffmpeg version 4.1.6 Copyright (c) 2000-2020 the FFmpeg developers built with gcc 7.5.0 (GCC)
> 
> @rwswan84, the problem is homebridge-camera-ffmpeg plugin uses ffmpeg, that's been preinstalled on your Synology NAS. I've just faced the same problem like you after updating to DSM 7 and switching from homebridge that had been running inside Docker container under DSM 6 to a native DSM 7 homebridge app.
> 
> So the solution you need is to manually update your outdated FFmpeg 4.1.6 to the latest 4.4.1
> 
> Here you can get a static FFmpeg build: https://www.johnvansickle.com/ffmpeg/ I have a Synology DS918+, which is Intel based, so I had to use amd64 version. Choose the one appropriate to your Synology NAS device. And here is the FAQ in case you are not much familiar with shell: https://www.johnvansickle.com/ffmpeg/faq/
> 
> Long story short - you need to turn on ssh in you Synology device's settings, connect with admin credentials and run the following commands (relevant on 6th of January 2022 for Intel-based devices): $ wget https://johnvansickle.com/ffmpeg/builds/ffmpeg-4.4.1-amd64-static.tar.xz $ tar xvf ffmpeg-release-amd64-static.tar.xz $ sudo mv /usr/bin/ffmpeg ./ $ sudo cp ./ffmpeg-4.4.1-amd64-static/ffmpeg /usr/bin/ $ chmod 755 /usr/bin/ffmpeg

Wow that worked! ffmpeg version in HomeBridge terminal now shows version 4.4.1-static 
I wonder who would be responsible for updating the version of ffmpeg that comes with the HomeBridge native app?

**Comment by Snoomy** [comment]
Happy to help. ) Exactly. That’s what I was trying to say: ffmpeg is being used for some other Synology apps like Surveillance Station for instance, so ffmpeg updates should come as the parts of DSM updates.

**Comment by rwswan84** [comment]
> 

Quick follow up question. Do you know how to fix this error in HomeBridge? I assume it's related to how the ffpmeg library is installed on Syngology
![image](https://user-images.githubusercontent.com/6299968/167960340-8abab390-b4a8-465d-8dfb-068008d08110.png)

**Comment by Sunoo** [comment]
@rwswan84 Talk to dgreif, that’s coming from the Ring plugin.

**Comment by rwswan84** [comment]
> 

The ring plugin uses the ffmpeg library for video/audio. It's saying that it found ffmpeg, but not the libfdk_aac library that is used for audio. So really this comes down to how ffmpeg is installed on Synology

**Comment by Sunoo** [comment]
The Ring plugin has nothing to do with this plugin.

Ask for help there: https://github.com/dgreif/ring

Or at the FFmpeg version used: https://github.com/homebridge/ffmpeg-for-homebridge

**Comment by rwswan84** [comment]
So the error is coming from the Ring plugin but the Ring plugin doesn't use ffmpeg? I'm confused

**Comment by Sunoo** [comment]
The Ring plugin has nothing to do with this plugin. Open an issue in either place I suggested.
